variables.tf

```diff
@@ -175,3 +176,45 @@ variable "logs_retention_days" {
   description = "Number of days to keep logs on storage account"
   default     = 30
 }
+
+variable "tdp_enabled" {
```

Comment:
Why not implement a single `threat_detection_policy` variable with object type and default value set to `null` ?